statsrunner/loop.py

```diff
@@ -68,47 +100,71 @@ def process_stats_element(ElementStats, tagname=None):
+
+                    tagname: Label for type of stats_module.
+                """
                 file_out = process_stats_file(FileStats)
                 out = process_stats_element(ElementStats, tagname)
-                return {'file':file_out, 'elements':out}
+                return {'file': file_out, 'elements': out}
 
             if root.tag == 'iati-activities':
                 stats_json = process_stats(stats_module.ActivityFileStats, stats_module.ActivityStats, 'iati-activity')
             elif root.tag == 'iati-organisations':
                 stats_json = process_stats(stats_module.OrganisationFileStats, stats_module.OrganisationStats, 'iati-organisation')
             else:
-                stats_json = {'file':{'nonstandardroots':1}, 'elements':[]}
+                stats_json = {'file': {'nonstandardroots': 1}, 'elements': []}
 
+    # If there is a ParseError print statement, then set stats_json file
+    # value according to whether the file size is zero.
     except etree.ParseError:
         print 'Could not parse file {0}'.format(inputfile)
         if os.path.getsize(inputfile) == 0:
             # Assume empty files are download errors, not invalid XML
-            stats_json = {'file':{'emptyfile':1}, 'elements':[]}
+            stats_json = {'file': {'emptyfile': 1}, 'elements': []}
         else:
-            stats_json = {'file':{'invalidxml':1}, 'elements':[]}
+            stats_json = {'file': {'invalidxml': 1}, 'elements': []}
 
+    # If args.verbose_loop is true, assign value of list of stats_json element
+    # keys to stats_json elements key and write to json file.
     if args.verbose_loop:
         with open(outputfile, 'w') as outfp:
             stats_json['elements'] = list(stats_json['elements'])
             json.dump(stats_json, outfp, sort_keys=True, indent=2, default=decimal_default)
+    # If args.verbose_loop is not true, create aggregated-file json and return
+    # the subtotal dictionary of statsrunner.aggregate.aggregate_file().
     else:
         statsrunner.aggregate.aggregate_file(stats_module, stats_json, os.path.join(output_dir, 'aggregated-file', folder, xmlfile))
 
 
 def loop_folder(folder, args, data_dir, output_dir):
+    """Given a folder, returns a list of XML files in folder."""
     if not os.path.isdir(os.path.join(data_dir, folder)) or folder == '.git':
         return []
     files = []
     for xmlfile in os.listdir(os.path.join(data_dir, folder)):
         try:
-            files.append((os.path.join(data_dir,folder,xmlfile),
+            files.append((os.path.join(data_dir, folder, xmlfile),
                          output_dir, folder, xmlfile, args))
         except UnicodeDecodeError:
             traceback.print_exc(file=sys.stdout)
             continue
     return files
 
+
 def loop(args):
+    """Loops through all specified folders to convert data to JSON output.
+
+    Args:
+        args: See __init__ for current defaults for statsrunner.
```

Comment:
When a program is run, an `args` variable is created containing the arguments that were passed into the program. For example, running* `run_this.py --bob 7 --jim happy` from the command line would lead to something like the following being available within code:
```
> args.bob
7
> args.jim
'happy'
```
It is then possible (though ill-advised) to modify the contents of `args` over the course of program execution or use the variable name for something else. My question is basically asking whether `args` has been modified within the code at all.

*syntax may be slightly off - haven't tested it